R/EXADBI-connection.R

```diff
@@ -0,0 +1,393 @@
+  "dbConnect", "EXADriver",
+  definition = function(drv, # change defaults also below
+                        exahost = "",
+                        uid = "",
+                        pwd = "",
+                        schema = "SYS",
+                        exalogfile = tempfile(pattern = "EXAODBC_", fileext = ".log"),
+                        logmode = "NONE",
+                        encryption = "N",
+                        autocommit = "Y",
+                        querytimeout = "0",
+                        connectionlcctype = Sys.getlocale(category = "LC_CTYPE"),
+                        connectionlcnumeric = Sys.getlocale(category = "LC_NUMERIC"),
+                        ...,
+                        dsn = "",
+                        connection_string = "")
+  {
+    .EXANewConnection(
+      drv = drv,
+      exahost = exahost,
+      uid = uid,
+      pwd = pwd,
+      schema = schema,
+      exalogfile = exalogfile,
+      logmode = logmode,
+      encryption = encryption,
+      autocommit = autocommit,
+      querytimeout = querytimeout,
+      connectionlcctype = connectionlcctype,
+      connectionlcnumeric = connectionlcnumeric,
+      ... = ...,
+      dsn = dsn,
+      connection_string = connection_string
+    )
+  },
+  valueClass = "EXAConnection"
+)
+
+#' @describeIn dbConnect Creates a connection to an EXASOL Database based on string.
+#' @family EXADriver related objects
+#' @family EXAConnection related objects
+#'
+#' @param drv Driver as character strink which is being forwared to \code{\link{dbDriver}}.
+#' @author EXASOL AG <opensource@exasol.com>
+#' @export
+setMethod(
+  "dbConnect", "character",
+  definition = function(drv, ...)
+    .EXANewConnection(drv = dbDriver(drv), ...),
+  valueClass = "EXAConnection"
+)
+
+#' @describeIn dbConnect Creates a connection to an EXASOL Database based on an existing connection.
+#' @family EXADriver related objects
+#' @family EXAConnection related objects
+#
+#' @param drv Existing connection
+#' @author EXASOL AG <opensource@exasol.com>
+#' @export
+setMethod(
+  "dbConnect", "EXAConnection",
+  definition = function(drv, ...)
+    .EXACloneConnection(drv, ...),
+  valueClass = "EXAConnection"
+)
+
+
+#' Fetches and outputs the current schema from an EXASOL DB. Also updates EXAConnection metadata.
+#' @family EXAConnection related objects
+#'
+#' @name dbCurrentSchema
+#' @param con a valid EXAConnection
+#' @param setSchema If not NULL, this schema will be opened and attached to EXAConnection metadata.
+#' @return an updated EXAConnection
+#' @export
+dbCurrentSchema <- function(con, setSchema=NULL) {
+  if(!missing(setSchema)) {
+    sqlQuery(con, paste("open schema", processIDs(setSchema)))
+    con@current_schema <- setSchema
+  } else {
+    res <- sqlQuery(con, "select current_schema")
+    con@current_schema <- as.character(res[1,1])
+  }
+  message(paste("Schema: ", con@current_schema))
+  con
+}
+
+.EXANewConnection <- function(# change defaults also above
```

Comment:
The name of this function has changed: The leading dot  marks as function as package internal.